### ISSUE TITLE
ATB error

### ISSUE THREAD

**Issue by rougier** [rougier/atb-python#1]
Citing Todd Marshall (unable to contact him):

I just was trying to use your atb python wrapper for AntTweakBar on Ubuntu 64bit 12.10. I downloaded your

http://www.loria.fr/~rougier/downloads/atb-python.tgz

directly from you and ran "sudo setup.py install". I tried your demo_glut.py and it crashed complaining about glumpy.key not existing. Maybe the newer version of glumpy used to have glumpy.key and has since moved it to glumpy.window.key etc

I had to make two simple changes in handlers_glut.py to get the demo_glut.py to run.

add "window" to achieve the following everywhere

 glumpy.key --> glumpy.window.key
 glumpy.mouse ---> glumpy.window.mouse

in VI I just did the following substitutions
%s/<glumpy.key>\C/glumpy.window.key/gp
%s/<glumpy.mouse>\C/glumpy.window.mouse/gp

Also, in demo_glut.py had to change

import trackball

to

import atb.trackball as trackball

**Comment by pocdn** [comment]
This is Todd Marshall. Thanks for opening this issue. In the meantime, I made an account here ...

**Comment by rougier** [comment]
Yep, I feared not being able to contact you since the mailer daemon bounced back my anwser.

atb-python is currently broken because glumpy should not appear at all in source, it is supposed to be self-contained and only use GLUT. Do you have a fix by any chance ?

**Comment by pocdn** [comment]
A possible fix is to just remove glumpy in handlers_glut.py and set the keyboard/mouse mappings directly.  Here's my version that does that; Note I chose to use an arbitrary naming convention of "_local_" but it may make sense to be something like "_pyatb_glut" to better delineate what this is for or if someone wants to add bindings to GLF/SDL/etc later.

#!/usr/bin/env python

# -_\- coding: utf-8 -_-

#-----------------------------------------------------------------------------

# Copyright (C) 2009-2010  Nicolas P. Rougier

#

# Distributed under the terms of the BSD License. The full license is in

# the file COPYING, distributed as part of this software.

#-----------------------------------------------------------------------------
import string
from raw import *

_local_key_map = {
 32: 32, #ord(' ')
 45: 45, #ord('-')
 46: 46, #ord('.')
 48: 48, #"0"
 49: 49, #1
 50: 50, #2
 51: 51, #3
 52: 52, #4
 53: 53, #5
 54: 54, #6
 55: 55, #7
 56: 56, #8
 57: 57, #'9'
 97: 65, #'a''A'
 98: 66,
 99: 67,
 100: 68,
 101: 69,
 102: 70,
 103: 71,
 104: 72,
 105: 73,
 106: 74,
 107: 75,
 108: 76,
 109: 77,
 110: 78,
 111: 79,
 112: 80,
 113: 81,
 114: 82,
 115: 83,
 116: 84,
 117: 85,
 118: 86,
 119: 87,
 120: 88,
 121: 89,
 122: 90, #'z''Z'
 65288: 8, #'\b'
 65293: 13
}

_local_button_map = {
 2: TW_MOUSE_LEFT, 
 4: TW_MOUSE_MIDDLE, 
 8: TW_MOUSE_RIGHT
}

_local_window_key_MOD_SHIFT = 1
_local_window_key_MOD_CTRL = 2
_local_window_key_MOD_ALT =4 

def map_key(key):
    return _local_key_map[key]

def map_button(button):
    return _local_button_map[button]

def map_modifiers(modifiers):
    ret = TW_KMOD_NONE
    if modifiers & _local_window_key_MOD_SHIFT:
        ret |= TW_KMOD_SHIFT
    if modifiers & _local_window_key_MOD_CTRL:
        ret |= TW_KMOD_CTRL
    if modifiers & _local_window_key_MOD_ALT:
        ret |= TW_KMOD_ALT
    return ret

class Handlers(object):

```
def __init__(self, window):
    self.window = window

def on_resize(self, width, height):
    TwWindowSize(width, height)

def on_key_press(self, symbol, modifiers):
    try:
        TwKeyPressed(map_key(symbol), map_modifiers(modifiers))
        self.window.draw()
        return True
    except:
        pass 
    return False
def on_mouse_press(self, x, y, button):
    if not button in _local_button_map.keys():
        return False
    if TwMouseButton(TW_MOUSE_PRESSED, map_button(button)):
        self.window.draw()
        return True

def on_mouse_release(self, x, y, button):
    if not button in _local_button_map.keys():
        return False
    if TwMouseButton(TW_MOUSE_RELEASED, map_button(button)):
        self.window.draw()
        return True

def on_mouse_drag(self, x, y, dx, dy, buttons):
    if TwMouseMotion(x, self.window.height-y):
        self.window.draw()
        return True

def on_mouse_motion(self, x, y, dx, dy):
    if TwMouseMotion(x, self.window.height-y):
        self.window.draw()
        return True

def on_draw(self):
    TwDraw()
```

**Comment by rougier** [comment]
Thanks. Can you make a pull request instead ?
